### PR TITLE
feature: Migrated input components to TS -> #19122

### LINKED ISSUE #19122: Migrate components to TS: Input
### Description
Migrate `Input` component to TypeScript

### Technical Details

- Convert Objects in `SNAKE_CASE` to enums in `PascalCase` (name and keys)
- Ensure enums use singular not plural e.g. `Size` not `Sizes`
- Update file name from `component-name.contants.js` to `component-name.types.ts` and add types
- Update file name from `component-name.js` to `component-name.tsx` and add types
- Update file name from `component-name.test.js` to `component-name.test.tsx`
- Update file name from `component-name.stories.js` to `component-name.stories.tsx`
- Update all instances of component objects to enums
- Update all documentation

### Acceptance Criteria

- Component has been converted to TypeScript
- Stories, documentation and tests have been updated inside of the `component-name` folder
- Before/After screen cast of component stories in storybook to show no visual regression after migration

### Difficulty: Intermediate

Good first issue for: External contributors who are familiar with TypeScript and want to contribute to improving the stability of the extension

### PR DESCRIPTION
**Pull request by git-init-priyanshu**
### Fixes #19122
**JavaScript to TypeScript Migration Completed Successfully**
I have successfully migrated all the JavaScript files to TypeScript files, following the provided instructions. By making this transition, we can now benefit from enhanced code maintainability, reliability, and scalability.

**Before**

**After**
image

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Comment by git-init-priyanshu** [comment]
I have read the CLA Document and I hereby sign the CLA

**Review by georgewrmarshall — CHANGES_REQUESTED** [review]
Hey @git-init-priyanshu, thanks for your contribution. You have instances of the incorrect casing for the enum which should be causing errors e.g. `InputType.text` should be `InputType.Text`. There are also a few other conventions you are missing. I would suggest having a look at the `BadgeWrapper` component. You also have a lot of failing tests including lint issues.

**Comment by git-init-priyanshu** [comment]
Hey @georgewrmarshall  thanks for reviewing my PR. 
I apologize for the errors and oversights in my recent contribution. I will make the necessary changes and review the BadgeWrapper component thoroughly to address any conventions I might have missed.

File: ui/components/component-library/input/index.ts
Comment (georgewrmarshall) [inline ui/components/component-library/input/index.ts:2]: Need a line break here
```suggestion
export { InputType } from './input.types';

```

File: ui/components/component-library/input/input.types.ts
Comment (georgewrmarshall) [inline ui/components/component-library/input/input.types.ts:6]: Need a line break here. As well as the `InputProps`
```suggestion
}
```

File: ui/components/component-library/input/input.tsx
Comment (georgewrmarshall) [inline ui/components/component-library/input/input.tsx:162]: You need to migrate the PropTypes to an interface in the `.types.ts` file

**Review by georgewrmarshall — COMMENTED** [review]
Still a bit to go :)

**Comment by git-init-priyanshu** [comment]
Thanks @georgewrmarshall for being so kind and patient, and for reviewing my PR. I will make sure to implement all these suggestions.

**Comment by git-init-priyanshu** [comment]
Made some tweaks but not sure if they are correct or not.

**Comment by git-init-priyanshu** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by git-init-priyanshu** [comment]
Hey @georgewrmarshall, could you please review my PR.
Thanks.

**Comment by georgewrmarshall** [comment]
Hey @git-init-priyanshu , Thanks for your time and effort in creating this pull request. As you're probably aware we have identified an issue related to the typing of the `Box` component, specifically with the polymorphic `as` prop. To address this issue, we have an open ticket at https://github.com/MetaMask/metamask-extension/issues/19239 and a draft pull request at https://github.com/MetaMask/metamask-extension/pull/19363 that needs to be merged before we can proceed with your PR. Thanks for your patience

**Comment by git-init-priyanshu** [comment]
Hey @georgewrmarshall, thanks for your reply
I'll wait for the necessary changes to be addressed and merged so that we can proceed smoothly. Once those changes are merged, I'll be ready to rebase and update my pull request accordingly.

**Comment by georgewrmarshall** [comment]
Blocked by https://github.com/MetaMask/metamask-extension/pull/19572

**Comment by georgewrmarshall** [comment]
Hey @git-init-priyanshu, thank you very much for your contribution. Another PR was opened that solves this task and it has one approval so I am going to close this one.

**Pull request closed without merging**
